src/Microsoft.AspNet.Mvc.Xml/Formatters/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -106,7 +126,7 @@ protected virtual XmlReader CreateXmlReader([NotNull] Stream readStream)
         /// <returns>The <see cref="XmlObjectSerializer"/> used during deserialization.</returns>
         protected virtual XmlObjectSerializer CreateDataContractSerializer(Type type)
         {
-            return new DataContractSerializer(SerializableErrorWrapper.CreateSerializableType(type));
```

Comment:
We have a PR to make this a full extensibility point #1872 feel free to comment there with ideas